notifier/src/main/kotlin/modules/JiraNotifier.kt

```diff
@@ -0,0 +1,140 @@
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.notifier.modules
+
+import com.atlassian.jira.rest.client.api.JiraRestClient
+import com.atlassian.jira.rest.client.api.RestClientException
+import com.atlassian.jira.rest.client.api.domain.BasicIssue
+import com.atlassian.jira.rest.client.api.domain.Comment
+import com.atlassian.jira.rest.client.api.domain.IssueType
+import com.atlassian.jira.rest.client.api.domain.input.IssueInputBuilder
+import com.atlassian.jira.rest.client.internal.async.AsynchronousJiraRestClientFactory
+
+import java.net.URI
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentMap
+
+import org.ossreviewtoolkit.model.Failure
+import org.ossreviewtoolkit.model.Result
+import org.ossreviewtoolkit.model.Success
+import org.ossreviewtoolkit.model.config.JiraConfiguration
+import org.ossreviewtoolkit.utils.collectMessagesAsString
+import org.ossreviewtoolkit.utils.log
+
+class JiraNotifier(
+    val config: JiraConfiguration,
+    private val restClient: JiraRestClient = AsynchronousJiraRestClientFactory()
+        .createWithBasicHttpAuthentication(URI(config.host), config.username, config.password)
+) {
+    /** Concurrent map of all available issue types for each project */
+    private val projectIssueTypes: ConcurrentMap<String, Map<String, IssueType>> = ConcurrentHashMap()
+
+    /**
+     * A Wrapper function in order to create an issue in the given Jira instance using the Jira REST API.
+     *
+     * @param projectKey The project key of the project in Jira in which the issue should be created.
+     * @param summary The summary of the issue.
+     * @param description The description of the issue.
+     * @param issueType The issue type of the issue. The given type has to be specified in the given project in Jira.
+     * @param assignee The assignee name of the issue. The given name has to be present in the Jira instance.
+     * @param avoidDuplicates If true (default), the function will not create a new issue if there is already exactly
+     *                        one issue with the same summary, but it will add a comment to the issue with the given
+     *                        description. If false, it will always create a new issue even if there are issues with the
+     *                        same summary.
+     *
+     * @return Return a [Result] that encapsulates the created issue on success, or the caught exception on failure.
+     */
+    fun createIssue(
```

Comment:
I used the `@param` tags on purpose because the parameter list is long and the use of the tags can still be found in other places in the code. Should I rewrite it or keep it with the the tags for readability?

Reply:
We do not have strict rule for this, but as the `@param` style is only found in three other files in the source code, and so the vast majority of files use the leaner KDoc format, I personally prefer to align on using the more commonly used lean KDoc format.